### PR TITLE
Daq d2scan, n-dimensional daq_anscan, daq_dnscan

### PR DESCRIPTION
**Pull request by tangkong**
## Description
- adds daq_d2scan, daq_anscan, daq_dnscan
- adds tests for modules

## Motivation and Context
[Jira Ticket](https://jira.slac.stanford.edu/browse/LCLSECSD-97)

## How Has This Been Tested?
Tested in hutch-python simulated mode on clone of pcds dev environment.  
Confirmed plans appear in `bp` namespace

## Where Has This Been Documented?
Docstrings have been updated.

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on travis
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Review by ZLLentz — COMMENTED** [review]
There isn't even anything to nitpick here, looks like a seamless extension of the existing scans. The tests should be sufficient like the equivalent tests are for the other scans. Hopefully the scientists appreciate the additional shortcuts provided here.

One small thing we do on a couple repositories is accumulate release notes on each PR instead of doing a deep dig on tag days. Can you run the docs/pre-release.sh script and fill in your description and github username on the form?

File: docs/source/upcoming_release_notes/49-daq_d2scan.rst [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+--------
+- N/A
+
+Maintenance
+-----------
+- N/A
+
+Contributors
+------------
+- roberttk
```
Comment (ZLLentz) [inline docs/source/upcoming_release_notes/49-daq_d2scan.rst:23]: We've been making these github usernames by convention for proper call-outs in the tag
```suggestion
- tangkong
```

Comment (tangkong) [reply docs/source/upcoming_release_notes/49-daq_d2scan.rst:23]: I actually forgot that was my github username... thanks.

**Review by ZLLentz — APPROVED** [review]

**Comment by klauer** [comment]
First PCDS PR from @tangkong  🎆 🍾 🥳

**Pull request merged**